plugins/vli/fu-vli-pd-device.c

```diff
@@ -241,6 +242,31 @@ fu_vli_pd_device_reset_vl103 (FuVliDevice *device, GError **error)
 					      NULL, error);
 }
 
+static gboolean
+fu_vli_pd_device_parade_setup (FuVliPdDevice *self, GError **error)
+{
+	g_autoptr(FuDevice) dev = NULL;
+	g_autoptr(GError) error_local = NULL;
+
+	/* add child */
+	dev = fu_vli_pd_parade_device_new (FU_VLI_DEVICE (self));
+	if (!fu_device_probe (dev, error))
+		return FALSE;
+	if (!fu_device_setup (dev, &error_local)) {
+		if (g_error_matches (error_local,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND)) {
+			g_debug ("%s", error_local->message);
+		} else {
+			g_warning ("cannot create I²C parade device: %s",
+				   error_local->message);
+		}
+		return TRUE;
```

Comment:
Do you actually want to return true in this instance?  I think you've got the return for probe and setup mixed up.  If probing fails, return TRUE, but if setup fails return FALSE is what makes sense to me.

Reply:
The way (I think) this works is that the PD chip is the parent, i.e. FuVliPdDevice and the parade device would be an optional child device. If the child is not found, we still want to be able to update the parent device.

Reply:
But why would you set the custom flag that it has the parade device if it wasn't true?   That seems like a hardcoded relationship.

Reply:
> I think you've got the return for probe and setup mixed up. 

Ohh I see what you mean, sorry. Fixed.